src/main/kotlin/platform/mixin/expression/MEExpressionCompletionUtil.kt

```diff
@@ -1179,6 +1178,11 @@ object MEExpressionCompletionUtil {
 
     private fun addDefinition(context: InsertionContext, id: String, definitionValue: String): PsiAnnotation? {
         val contextElement = context.file.findElementAt(context.startOffset) ?: return null
+        return addDefinition(contextElement, id, definitionValue)
+    }
+
+    fun addDefinition(contextElement: PsiElement, id: String, definitionValue: String): PsiAnnotation? {
+        val project = contextElement.project
```

Comment:
We should keep the project as a parameter if possible, as IIRC accessing the project from a PSI element is less performant. Although it probably doesn't matter here.

Reply:
It got the project from the element prior to my changes

Reply:
Oh sorry it was from the `InsertionContext`, I see.